GPUQREngine/Source/BucketList/BucketList.cpp

```diff
@@ -105,13 +106,31 @@ BucketList::BucketList
     }
 }
 
-BucketList::~BucketList()
+template BucketList<int32_t>::BucketList
+(
+    Front <int32_t> *F,
+    int32_t minApplyGranularity
+) ;
+template BucketList<int64_t>::BucketList
+(
+    Front <int64_t> *F,
+    int64_t minApplyGranularity
+) ;
```

Comment:
Is it really necessary to explicitly instantiate each member function? Wouldn't it be enough to just instantiate the class template?

I.e., after all (templated) member functions and possible specializations are defined, just add the following two lines to this file:
```
template class BucketList<int32_t>;
template class BucketList<int64_t>;
```